src/mjbots_robot_interface.cpp

```diff
@@ -47,28 +47,19 @@ ::mjbots::moteus::QueryResult MjbotsRobotInterface::Get(const std::vector<::mjbo
   return {};
 }
 
-MjbotsRobotInterface::MjbotsRobotInterface(const std::vector<JointMoteus> &joint_list,
+MjbotsRobotInterface::MjbotsRobotInterface(std::vector<std::shared_ptr<JointMoteus>> joint_ptrs,
                                            const RealtimeParams &realtime_params,
-                                           int soft_start_duration,
-                                           float robot_max_torque,
                                            ::mjbots::pi3hat::Euler imu_mounting_deg,
-                                           int imu_rate_hz) :
-                                            soft_start_(robot_max_torque, soft_start_duration) { 
-
-  for (JointMoteus joint: joint_list){
-    // Make vector of shared_pointer objects using joint copy constructer 
-    // (original list will be descoped)
-    joints.push_back(std::make_shared<JointMoteus>(joint)); 
-  }
+                                           int imu_rate_hz) 
+  { 
+  joints = joint_ptrs;
+  num_joints_ = joints.size();
 
   for( auto & j: joints){
-    positions_.push_back( j->get_position_reference() );
-    velocities_.push_back( j->get_velocity_reference() );
-    torque_cmd_.push_back( j->get_servo_torque_reference()   ); 
     modes_.push_back(j->get_mode_reference());
   }
-  num_servos_ = joints.size();
-  for (size_t i = 0; i < num_servos_; ++i)
+  
+  for (size_t i = 0; i < num_joints_; ++i)
```

Comment:
Changed for consistency still num_motors in controlLoop, should be the same everywhere?

Reply:
Should be consistent everywhere, also `num_joints` is more apt.

Reply:
Resolved in 8d4fd951d65b7cc23aedfa556c2617725146633e.